File: BOJ/Python/9663-N-Queen.py
n = int(input())
def calc(stack=[]):
    global cnt
    if len(stack) == n:
        cnt += 1
        return
    for i in range(n):
        if i in stack:
            continue

        flag = False
        for j in range(len(stack)):
            if j+1 == abs(stack[-j-1]-i):
                flag = True
                break
        if flag:
            continue
        calc(stack+[i])
cnt = 0
calc()
print(cnt)

File: BOJ/C++/9663-N-Queen.cpp
#include <iostream>
int n, cnt;
void calc(int &stack, int qC) {
    if(qC == n) {
        cnt += 1;
        return;
    }
    for(int i = 0; i < n; i++) {
        for(int j = 0; j < qC; j++) {
            if(stack[j] == i) {
                continue;
            }
        }

    }
}

int main() {
    std::cin >> n;
    cnt = 0
}